### PR TITLE
Benchmark fix for warp reduced softmax

### LINKED ISSUE #1194: Softmax warp reduce benchmarks broken
Getting the following error on `Softmax_WarpReduce` benchmark.
```
terminate called after throwing an instance of 'c10::Error'
  what():  *val <= padded_constant_it->secondINTERNAL ASSERT FAILED at "../torch/csrc/jit/codegen/cuda/executor.cpp":452, please report a bug to PyTorch. 
Exception raised from computeLaunchParams at ../torch/csrc/jit/codegen/cuda/executor.cpp:452 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x6b (0x7fa653ec581b in /home/csarofeen/pytorch/build/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, char const*) + 0xd1 (0x7fa653ec15f1 in /home/csarofeen/pytorch/build/lib/libc10.so)
frame #2: torch::jit::fuser::cuda::FusionExecutor::computeLaunchParams(torch::jit::fuser::cuda::LaunchParams const&, torch::jit::fuser::cuda::kir::ExpressionEvaluator&) + 0xad1 (0x7fa6575f1201 in /home/csarofeen/pytorch/build/lib/libtorch_cuda.so)
frame #3: torch::jit::fuser::cuda::FusionExecutor::runFusion(c10::ArrayRef<c10::IValue> const&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, torch::jit::fuser::cuda::LaunchParams const&, c10::optional<unsigned long> const&) + 0x3d2 (0x7fa6575f23a2 in /home/csarofeen/pytorch/build/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0x77c8f (0x55aafb6a6c8f in ./build/bin/nvfuser_bench)
frame #5: benchmark::internal::BenchmarkInstance::Run(unsigned long, int, benchmark::internal::ThreadTimer*, benchmark::internal::ThreadManager*, benchmark::internal::PerfCountersMeasurement*) const + 0xb0 (0x55aafb7057d0 in ./build/bin/nvfuser_bench)
frame #6: <unknown function> + 0xb2aaa (0x55aafb6e1aaa in ./build/bin/nvfuser_bench)
frame #7: benchmark::internal::BenchmarkRunner::DoNIterations() + 0x373 (0x55aafb6e2243 in ./build/bin/nvfuser_bench)
frame #8: benchmark::internal::BenchmarkRunner::DoOneRepetition() + 0x10b (0x55aafb6e293b in ./build/bin/nvfuser_bench)
frame #9: <unknown function> + 0x8dfd2 (0x55aafb6bcfd2 in ./build/bin/nvfuser_bench)
frame #10: benchmark::RunSpecifiedBenchmarks(benchmark::BenchmarkReporter*, benchmark::BenchmarkReporter*) + 0x620 (0x55aafb6be750 in ./build/bin/nvfuser_bench)
frame #11: main + 0x30 (0x55aafb65d8b0 in ./build/bin/nvfuser_bench)
frame #12: __libc_start_main + 0xe7 (0x7fa65327cb97 in /lib/x86_64-linux-gnu/libc.so.6)
frame #13: _start + 0x2a (0x55aafb65e1ca in ./build/bin/nvfuser_bench)
```

Wonder if this is related to https://github.com/csarofeen/pytorch/pull/1193 or https://github.com/csarofeen/pytorch/pull/1073

### PR DESCRIPTION
**Pull request by shmsong**
Fixes #1194 .

Persistent scheduler updated and the benchmark couldn't statically assume single-warp reduce anymore. Just removed that from the benchmark. 

Also added this case to cpp test so it could be detected early.

### CODE REVIEW

**Review by csarofeen — APPROVED** [review]
LGTM, Thanks for the fix!
